File: Day-8/MinSum.go
import (
	"fmt"
	"sort"
	"strings"
)

func minSum(arr []int) string {
	sort.Ints(arr)

	var num1, num2 strings.Builder

	for i, val := range arr {
		if i%2 == 0 {
			num1.WriteString(fmt.Sprintf("%d", val))
		} else {
			num2.WriteString(fmt.Sprintf("%d", val))
		}
	}

	// Remove leading zeros
	n1 := strings.TrimLeft(num1.String(), "0")
	n2 := strings.TrimLeft(num2.String(), "0")

	if n1 == "" {
		n1 = "0"
	}
	if n2 == "" {
		n2 = "0"
	}

	// Add large numbers as strings
	return addStrings(n1, n2)
}

// Add two large numbers represented as strings
func addStrings(a, b string) string {
	i, j := len(a)-1, len(b)-1
	carry := 0
	var res strings.Builder

	for i >= 0 || j >= 0 || carry > 0 {
		sum := carry
		if i >= 0 {
			sum += int(a[i] - '0')
			i--
		}
		if j >= 0 {
			sum += int(b[j] - '0')
			j--
		}
		res.WriteByte(byte(sum%10) + '0')
		carry = sum / 10
	}

	// Reverse the result
	runes := []byte(res.String())
	for l, r := 0, len(runes)-1; l < r; l, r = l+1, r-1 {
		runes[l], runes[r] = runes[r], runes[l]
	}

	return string(runes)
}

File: Day-8/MinSum.js
function minSum(arr) {
  arr.sort((a, b) => a - b);

  let num1 = '';
  let num2 = '';

  for (let i = 0; i < arr.length; i++) {
      if (i % 2 === 0)
          num1 += arr[i];
      else
          num2 += arr[i];
  }

  // Remove leading zeros
  num1 = num1.replace(/^0+/, '') || '0';
  num2 = num2.replace(/^0+/, '') || '0';

  return addStrings(num1, num2);
}

function addStrings(a, b) {
  let i = a.length - 1, j = b.length - 1;
  let carry = 0;
  let res = [];

  while (i >= 0 || j >= 0 || carry > 0) {
      let sum = carry;
      if (i >= 0) sum += +a[i--];
      if (j >= 0) sum += +b[j--];
      res.push(sum % 10);
      carry = Math.floor(sum / 10);
  }

  return res.reverse().join('');
}